### PR TITLE
[Chat] Group consecutive completed tool calls

### PR DESCRIPTION
**Pull request by wanglam**
### Description

This PR introduces tool call grouping functionality in the OpenSearch Dashboards chat interface. The feature groups completed tool calls from consecutive assistant messages into a single visual unit, improving UI clarity and reducing clutter when multiple tools execute in sequence.

## Key Changes

- **New Components**: Added `tool_call_group.tsx` and `tool_call_group.scss` for rendering grouped tool calls
- **Enhanced Logic**: Implemented `convertTimelineToMessageRows()` function that intelligently groups completed tool calls while displaying running tools individually
- **Comprehensive Testing**: Added 283 lines of unit tests covering grouping scenarios, edge cases, and error handling
- **UI Improvements**: Tool calls now display as grouped batches when all tools in a sequence complete, with individual display for in-progress operations

The grouping logic only activates when a batch of tool calls is complete (all have results and are followed by an assistant message), ensuring users see real-time progress for running operations.

## Screenshot

https://github.com/user-attachments/assets/3f0d9cdf-de12-480e-9f68-2ee4bce80c1d

## Changelog

- feat: [Chat]Group consecutive completed tool calls

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by SuZhou-Joe** [comment]
Seems changeset file is missing in this PR.

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/chat/public/components/tool_call_group.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          <EuiFlexItem grow={false}>
+            <EuiIcon size="m" type="checkInCircleEmpty" color="success" />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            {i18n.translate('chat.tooCallGroup.buttonLabel', {
+              defaultMessage: '{toolCount} tasks performed',
+              values: {
+                toolCount: toolCalls.length,
+              },
+            })}
```
[the remark this answers is not in the document]
Comment (ananzh) [reply src/plugins/chat/public/components/tool_call_group.tsx:31]: +1 missing `l`

File: src/plugins/chat/public/components/chat_messages.tsx [hunk 2]
```diff
@@ -63,6 +64,140 @@ interface ChatMessagesProps {
...
+    );
+
+    return {
+      type: 'tool_call' as const,
+      id: toolCall.id,
+      toolName: toolCall.function.name,
+      status: getToolStatus(toolCall, toolResult),
+      arguments: toolCall.function.arguments,
+      result: toolResult?.content,
+      timestamp: Date.now(),
```
Comment (ananzh) [inline src/plugins/chat/public/components/chat_messages.tsx:103]: does msg itself has a timestamp?

**Review by ananzh — APPROVED** [review]

File: src/plugins/chat/public/components/chat_messages.tsx (hunk 2 above)
Comment (wanglam) [reply src/plugins/chat/public/components/chat_messages.tsx:103]: Just copied from the previous implementation, forget to add the original comments.
